### PR TITLE
Move Python shebang hack to Komodo, make general

### PR DESCRIPTION
**Pull request by pgdr**

### CODE REVIEW

File: komodo/__init__.py [hunk 1]
```diff
@@ -1,12 +1,50 @@
...
+    This is a hack that should be fixed at some point.
+
+    """
+    # TODO fix hardcoded fix of ipython, bokeh, ...
+    binpath = os.path.join(args.prefix, args.release, 'root', 'bin')
+    python_ = os.path.join(binpath, 'python')
+
+    bins = []
+    # executables with wrong shebang
+    for bin_ in os.path.walk(binpath):
```
Comment (joakim-hove) [inline komodo/__init__.py]: I think this as on old - not anymore working syntax? As long as you do not intend to descend the filesystem recursively you could just us `os.listdir( )` - and irrespective  of which method is used I think there is a missing `os.path.join( )` in the `open( )`call.

File: komodo/__init__.py [hunk 2]
```diff
@@ -1,12 +1,50 @@
...
+            with open(bin_, 'r') as f:
+                shebang = f.readline()
+            if 'python' in shebang and shebang[2:] != binpath:
+                bins_.append(bin_)
+        except:
+            pass
+
+    sedfxp = """sed -i 1c#!{0} {1}"""
+    for bin_ in bins_:
+        binpath_ = os.path.join(args.prefix, args.release, 'root', 'bin', bin_)
```
Comment (joakim-hove) [inline komodo/__init__.py]: There are many different `binxxx` variables here, denoting directories, filenames and full paths to files; maybe some different names to make things clearer?

File: komodo/__init__.py [hunk 3]
```diff
@@ -1,12 +1,50 @@
...
+    deploys.  This breaks the application since the corresponding Python modules
+    won't be picked up correctly.
+
+    For now, we use sed to rewrite the first line in some executables.
+
+    This is a hack that should be fixed at some point.
+
+    """
+    # TODO fix hardcoded fix of ipython, bokeh, ...
+    binpath = os.path.join(args.prefix, args.release, 'root', 'bin')
```
Comment (joakim-hove) [inline komodo/__init__.py]: Should we replace one hardcoded hash-bang with another? We could alternatively replace with: `#!/usr/bin/env python`?

**Comment by joakim-hove** [comment]
I am very pleased to see development and improvements to this code!!

**Comment by pgdr** [comment]
Hi, @joakim-hove  ... I think this is ready for another look.  I've run it on `web` in `komodo-advanced` now, and the output is as expected (as far as I can tell)

```

> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/2to3
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/python-config
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/smtpd.py
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/futurize
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/pywhich
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/pip2.7
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/iptest
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/pip2
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/flask
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/idle
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/pygmentize
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/easy_install
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/pip
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/chardetect
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/f2py
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/python2-config
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/wheel
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/virtualenv
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/iptest2
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/easy_install-2.7
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/pasteurize
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/ipython
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/bokeh
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/ipython2
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/python2.7-config
> sed -i 1c#!/project/res/komodo/web/root/bin/python /project/res/komodo/web/root/bin/pydoc
```

**Pull request merged**